src/test/java/tennisclub/dao/LessonDaoTest.java

```diff
@@ -0,0 +1,350 @@
+package tennisclub.dao;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import tennisclub.entity.Court;
+import tennisclub.entity.Lesson;
+import tennisclub.entity.User;
+import tennisclub.entity.enums.Level;
+
+import javax.persistence.*;
+import javax.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * @author Miroslav Demek
+ */
+@SpringBootTest
+public class LessonDaoTest {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Autowired
+    private LessonDao lessonDao;
+
+    private Court court;
+
+    private User teacher;
+    private User student;
+
+    private Lesson lesson;
+
+    @BeforeEach
+    @Transactional
```

Comment:
It is enough to mark the whole class as `@Transactional` instead of annotating each method

Reply:
That way, also the helper method `createLessonFromTime` would be transactional, which it doesn't need to, but I suppose that's not too much of an issue.